### DISCUSSION TITLE
Long burst timelaspe- inconsistent inner clock and not able to set photo interval and start-stop times

### DISCUSSION THREAD

**Discussion by kelp-videos** [gopro/labs#740] [General]
Hi all, I am having 2 main issues with my GoPro labs script and looking for help:

1. I tried out this pre-program (in the attached photos) overnight to take a burst every 10 minutes. It seemed to record the correct data, but when I looked at the time stamps, the burst photos were taken in slightly greater intervals than 10 minutes. I was wondering if the time it takes to process the photos would slow down the internal clock? Also, is there an option to attach the gopro to an external clock if this issue cant be fixed internally?

2. I want to program my gopro to take 5 burst photos, every 10 minutes, between the times 22:00 and 7:30. When I tried to put both the 10 minute intervals and 22:00-7:30 in the presets, it only acted on the 10 minute interval programs. Could someone help with this issue too?
![Screenshot (51)](https://github.com/gopro/labs/assets/157651264/10089431-f86d-4bdd-b6e7-450e1f76d45e)
![Screenshot (52)](https://github.com/gopro/labs/assets/157651264/2e29e257-4eaf-4572-95a9-1d76bc1bcd2d)

**Comment by dnewman-gpsw** [comment]
This is not an issue (i.e. a confirmed bug), moving to discussions.

**Comment by dnewman-gpsw** [comment]
1) the command is performed in order. You asked in to sleep for 10 minutes, then you ask it to set a bunch of parameters, which take some time, then you start a capture, which takes time. So it doing what you asked, and is not a bug. If you want closer to 10 minutes between images, but find your internal is more like 615 seconds, you can set you sleep to be !575N (575 seconds rather than 600). With more recent models (H11-12) you can script to look at the clock. You didn't mention the camera model.

2) You will to learn the scripting language, that web page is just for simple actions.  If you on H11 or H12 you can use https://gopro.github.io/labs/build/ to code you our Labs scripts.

All the language docs are here: https://gopro.github.io/labs/control/tech/

A basic version that will run on mostly Labs enabled models: ```>07:30<22:00!22:00NmPB!S!575R```

```
if(time>=07:30 && time<22:00)   // >07:30<22:00
  Sleep until 22:00             // !22:00N
set Mode Photo Burst            // mPB
Start Capture                   // !S
Repeat in 575s                  // !575R
```
